packages/block-library/src/navigation/style.scss

```diff
@@ -76,14 +76,12 @@ $navigation-icon-size: 24px;
 		}
 	}
 
-	&:where(:not([class*="has-text-decoration"])) {
-		a {
-			text-decoration: none;
+	&:where(:not([class*="has-text-decoration"])) :where(a) {
```

Comment:
Could these be combined into a single `where`?

Reply:
Another question: do we still need the `:not([class*="has-text-decoration"])` if the specificity is lowered? Ideally if it has that class, it should just override this rule and the exception should no longer be needed.

Reply:
> Could these be combined into a single where?

I tried this:
```css
&:where(:not([class*="has-text-decoration"]) a)
```

But that selector doesn't work. Is there another way?

Reply:
> Another question: do we still need the :not([class*="has-text-decoration"]) if the specificity is lowered? Ideally if it has that class, it should just override this rule and the exception should no longer be needed.

It does seem needed, it overrides a general rule for links that are not buttons which gives them underlines:
![Screenshot 2024-07-12 at 3 24 33 PM](https://github.com/user-attachments/assets/e383c321-ee6c-4ee9-86be-5d9bfd997be7)

That should be a bit more obvious when testing now that I've rebased to include the fix from #63403.

Reply:
I've applied Aaron's suggestion now so it looks a little different - https://github.com/WordPress/gutenberg/pull/63406#discussion_r1675405046

Reply:
> It does seem needed, it overrides a general rule for links that are not buttons which gives them underlines:

Removing the `:where(:not([class*=has-text-decoration]))` won't change the specificity. So the selector would remain `0-1-0` specificity and continue to override `a:where(:not(.wp-element-button))` which is only `0-0-1`.

Reply:
Oh, I misunderstood what was meant in the original comment. I've gone ahead and removed that. 👍